src/ngraph/pass/constant_folding_scatter.cpp

```diff
@@ -0,0 +1,301 @@
+        return fold_constant_scatter_elem_updt<T, U, int32_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::i64:
+        return fold_constant_scatter_elem_updt<T, U, int64_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::boolean:
+    case element::Type_t::bf16:
+    case element::Type_t::f16:
+    case element::Type_t::f32:
+    case element::Type_t::f64:
+    case element::Type_t::u1:
+    default:
+        NGRAPH_CHECK(false,
+                     "Encountered unsupported axis element type in "
+                     "constant_scatter_elem_updt_callback: ",
+                     axis_type);
+        break;
+    }
+}
+
+template <typename T>
+static shared_ptr<op::Constant> dispatch_const_fold_data(const shared_ptr<op::Constant>& data,
+                                                         const shared_ptr<op::Constant>& indices,
+                                                         const shared_ptr<op::Constant>& updates,
+                                                         const shared_ptr<op::Constant>& axis,
+                                                         const shared_ptr<Node>& scatter_elem_updt)
+{
+    auto indices_type = indices->get_output_element_type(0);
+
+    // Dispatch specialization based on indicies data type.
+    switch (indices_type)
+    {
+    case element::Type_t::undefined:
+        NGRAPH_CHECK(false,
+                     "Encountered 'undefined' element type in constant_scatter_elem_updt_callback");
+        break;
+    case element::Type_t::dynamic:
+        NGRAPH_CHECK(false,
+                     "Encountered 'dynamic' element type in constant_scatter_elem_updt_callback");
+        break;
+    case element::Type_t::u8:
+        return dispatch_const_fold_indices<T, uint8_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::u16:
+        return dispatch_const_fold_indices<T, uint16_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::u32:
+        return dispatch_const_fold_indices<T, uint32_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::u64:
+        return dispatch_const_fold_indices<T, uint64_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::i8:
+        return dispatch_const_fold_indices<T, int8_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::i16:
+        return dispatch_const_fold_indices<T, int16_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::i32:
+        return dispatch_const_fold_indices<T, int32_t>(
+            data, indices, updates, axis, scatter_elem_updt);
+    case element::Type_t::i64:
+        return dispatch_const_fold_indices<T, int64_t>(
+            data, indices, updates, axis, scatter_elem_updt);
```

Comment:
You can lie can call these uint*_t and it will work fine. Same with bf16, f16, etc.

Reply:
Done. Please take a look if this is correct.